indico/modules/attachments/util.py

```diff
@@ -61,16 +61,19 @@ def get_attached_items(linked_object, include_empty=True, include_hidden=True, p
 
 def can_manage_attachments(obj, user, allow_admin=True):
     """Check if a user can manage attachments for the object."""
+    from indico.modules.attachments import attachments_settings
     if not user:
         return False
     if obj.can_manage(user, allow_admin=allow_admin):
         return True
     if isinstance(obj, db.m.Event) and obj.can_manage(user, 'submit', allow_admin=allow_admin):
-        return True
+        return not attachments_settings.get(obj, 'managers_only')
```

Comment:
hmmm why not just an `elif not ...: return False` after the can_manage check?

Reply:
I just did it this way to make use of the `isinstance` (since the settings proxy needs the event, and before that point we don't know the type of `obj`).
Otherwise we would have to do something like this to get the event:
```python
event = (obj
         if isinstance(obj, db.m.Event)
         else obj.event
         if isinstance(obj, db.m.Contribution)
         else obj.contribution.event
         if isinstance(obj, db.m.SubContribution)
         else None)
```

Reply:
All those objects  including events have `.event`, but I guess categories are the one exception... 

This should work though:

```python
event = getattr(obj, 'event', None)
if event and not attachments_settings.get(event, 'managers_only'):
    return False
```